tests/sample_qc/test_de_novo.py

```diff
@@ -0,0 +1,244 @@
+                hl.literal(True),
+                hl.literal(1e-5),
+                False,
+                hl.struct(
+                    is_de_novo=True,
+                    p_de_novo=hl.missing(hl.tfloat64),
+                    confidence=hl.missing(hl.tstr),
+                    fail_reason=hl.set(
+                        {
+                            "min_de_novo_p",
+                            "min_proband_ab",
+                            "min_proband_gq",
+                            "parent_sum_ad_0",
+                        }
+                    ),
+                ),
+            ),
+            # Case 2: One fail condition (low DP ratio)
+            (
+                hl.locus("chr1", 20000, reference_genome="GRCh38"),
+                hl.literal(["A", "T"]),
+                hl.struct(GT=hl.call(0, 1), AD=[20, 5], DP=10, GQ=50, PL=[10, 0, 100]),
+                hl.struct(GT=hl.call(0, 0), AD=[10, 0], DP=100, GQ=99, PL=[0, 99, 198]),
+                hl.struct(GT=hl.call(0, 0), AD=[10, 0], DP=100, GQ=99, PL=[0, 99, 198]),
+                hl.literal(False),
+                hl.literal(1e-5),
+                False,
+                hl.struct(
+                    is_de_novo=True,
+                    p_de_novo=hl.missing(hl.tfloat64),
+                    confidence=hl.missing(hl.tstr),
+                    fail_reason=hl.set({"min_dp_ratio"}),
+                ),
+            ),
+            # Case 3: Variant is inherited (not de novo)
+            (
+                hl.locus("chr1", 30000, reference_genome="GRCh38"),
+                hl.literal(["G", "T"]),
+                hl.struct(GT=hl.call(0, 1), AD=[15, 10], DP=30, GQ=50, PL=[10, 0, 100]),
+                hl.struct(GT=hl.call(0, 1), AD=[10, 5], DP=20, GQ=40, PL=[0, 20, 80]),
+                hl.struct(GT=hl.call(0, 0), AD=[20, 0], DP=20, GQ=50, PL=[0, 99, 198]),
+                hl.literal(True),
+                hl.literal(1e-5),
+                False,
+                hl.struct(
+                    is_de_novo=False,
+                    p_de_novo=hl.missing(hl.tfloat64),
+                    confidence=hl.missing(hl.tstr),
+                    fail_reason=hl.missing(hl.tset(hl.tstr)),
+                ),
+            ),
+            # Case 4: A passing case (high confidence de novo)
+            (
+                hl.locus("chr1", 40000, reference_genome="GRCh38"),
+                hl.literal(["C", "G"]),
+                hl.struct(GT=hl.call(0, 1), AD=[5, 30], DP=35, GQ=99, PL=[99, 0, 1]),
+                hl.struct(GT=hl.call(0, 0), AD=[20, 0], DP=20, GQ=60, PL=[0, 60, 120]),
+                hl.struct(GT=hl.call(0, 0), AD=[25, 0], DP=25, GQ=80, PL=[0, 80, 150]),
+                hl.literal(True),
+                hl.literal(1e-5),
+                False,
+                hl.struct(
+                    is_de_novo=True,
+                    p_de_novo=0.999,  # High confidence P(de novo)
```

Comment:
```suggestion
                    p_de_novo=0.999,
```

Reply:
it might be worth checking MEDIUM and LOW as well, and also including an indel test case. what do you think about including all three confidence levels?